src/main/java/mods/railcraft/common/util/inventory/InvTools.java

```diff
@@ -78,6 +78,34 @@ public static ItemStack emptyStack() {
         return null;
     }
 
+    /**
+     * This method changes the original stack!
+     */
+    @Nullable
+    public static ItemStack decSize(@Nullable ItemStack stack) {
```

Comment:
Most of the uses of `ItemStack#stackSize` field are direct mutations, so it is not that bad an idea. In this case, this stack is in player's hand, consuming one after dying the smoker.